### PR TITLE
Clarify what it means to be "unsupported" by a supports expression.

### PR DESCRIPTION
**Pull request by BillyONeal**
@JavierMatosD Hopefully this makes it clearer for the customer you were working with.

Also fixes a few places where we were printing a spec but should have only printed a package name.

### CODE REVIEW

**Review by JavierMatosD — APPROVED** [review]
Thank you!

**Comment by dg0yt** [comment]
What I find confusing is how the message prints the requested platform first, i.e.:
~~~~
atlmfc[core]:x64-linux is only supported on 'windows'
~~~~
... this reads like "The linux package is only supported on windows".
Not sure how to improve this concisely, maybe
~~~
atlmfc[core] is unsupported on x64-linux but supported on 'windows'.
...
~~~

(This area also needs improvements when a dependency is unsupported. ATM it doesn't tell the requested feature which asks for unsupported dependency. This is difficult to resolve.)

**Comment by BillyONeal** [comment]
> What I find confusing is how the message prints the requested platform first, i.e.:

That's fair, I'll take a look at this again

> This area also needs improvements when a dependency is unsupported.

I agree.

File: locales/messages.json [hunk 1]
```diff
@@ -1115,10 +1115,12 @@
   "_UnsupportedPort.comment": "An example of {package_name} is zlib.",
   "UnsupportedPortDependency": "- dependency {value} is not supported.",
   "_UnsupportedPortDependency.comment": "'{value}' is the name of a port dependency.",
-  "UnsupportedPortFeature": "{spec} is only supported on '{supports_expression}'",
-  "_UnsupportedPortFeature.comment": "An example of {spec} is zlib:x64-windows. An example of {supports_expression} is windows & !static.",
   "UnsupportedShortOptions": "short options are not supported: '{value}'",
   "_UnsupportedShortOptions.comment": "'{value}' is the short option given",
+  "UnsupportedSupportsExpression": "{spec} is only supported on '{supports_expression}'. This usually means that there are known build failures, or runtime problems, when building other platforms. To ignore this and attempt to build {spec} anyway, rerun vcpkg with `--allow-unsupported`.",
+  "_UnsupportedSupportsExpression.comment": "An example of {spec} is zlib:x64-windows. An example of {supports_expression} is windows & !static.",
```
Comment (Neumann-A) [inline locales/messages.json:1121]:
```suggestion
  "_UnsupportedSupportsExpression.comment": "An example of {spec} is zlib:x64-windows. An example of {supports_expression} is 'windows & !static'.",
```

File: locales/messages.json [hunk 2]
```diff
@@ -1115,10 +1115,12 @@
...
+  "UnsupportedSupportsExpressionWarning": "{spec} is only supported on '{supports_expression}'. This usually means that there are known build failures, or runtime problems, when building other platforms. Proceeding anyway due to `--allow-unsupported`.",
+  "_UnsupportedSupportsExpressionWarning.comment": "An example of {spec} is zlib:x64-windows. An example of {supports_expression} is windows & !static.",
```
Comment (Neumann-A) [inline locales/messages.json:1123]:
```suggestion
  "_UnsupportedSupportsExpressionWarning.comment": "An example of {spec} is zlib:x64-windows. An example of {supports_expression} is 'windows & !static'.",
```

File: locales/messages.json [hunk 3]
```diff
@@ -1115,10 +1115,12 @@
   "_UnsupportedPort.comment": "An example of {package_name} is zlib.",
   "UnsupportedPortDependency": "- dependency {value} is not supported.",
   "_UnsupportedPortDependency.comment": "'{value}' is the name of a port dependency.",
-  "UnsupportedPortFeature": "{spec} is only supported on '{supports_expression}'",
-  "_UnsupportedPortFeature.comment": "An example of {spec} is zlib:x64-windows. An example of {supports_expression} is windows & !static.",
   "UnsupportedShortOptions": "short options are not supported: '{value}'",
   "_UnsupportedShortOptions.comment": "'{value}' is the short option given",
+  "UnsupportedSupportsExpression": "{spec} is only supported on '{supports_expression}'. This usually means that there are known build failures, or runtime problems, when building other platforms. To ignore this and attempt to build {spec} anyway, rerun vcpkg with `--allow-unsupported`.",
```
Comment (ras0219-msft) [inline locales/messages.json:1120]:
```suggestion
  "UnsupportedSupportsExpression": "{package_name} is only supported on '{supports_expression}'. This usually means that there are known build failures, or runtime problems, when building other platforms. To ignore this and attempt to build {package_name} anyway, rerun vcpkg with `--allow-unsupported`.",
```

`{spec}` implies triplet information (see comment). I think `{package_name}` is the closest we have without defining a new placeholder.

Comment (BillyONeal) [reply locales/messages.json:1120]: We are printing triplet information here; that's @dg0yt's comment :)

Comment (dg0yt) [reply locales/messages.json:1120]: Package, features and triplets done right...
See the gdal example I added to https://github.com/microsoft/vcpkg-tool/pull/879#issuecomment-1411492640.
What I want is app.
~~~
$ ./vcpkg install gdal[*]:x64-linux
Computing installation plan...
gdal[aws-ec2-windows] cannot be installed:
atlmfc[core]:x64-linux is only supported on 'windows'
~~~

Comment (BillyONeal) [reply locales/messages.json:1120]: I agree that would be an improvement but that's more than the 10 minute "this should explain what we mean" fix I was going for here.

File: locales/messages.json (hunk 1 above)
Comment (BillyONeal) [reply locales/messages.json:1121]: This (and all other examples) are generated code.

File: include/vcpkg/base/messages.h [hunk 4]
```diff
@@ -2213,9 +2223,10 @@ namespace vcpkg
                     "",
                     "To remove dependencies in manifest mode, edit your manifest (vcpkg.json) and run 'install'.");
     DECLARE_MESSAGE(RemovePackageConflict,
```
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:2225]: image

**Review by JavierMatosD — APPROVED** [review]

**Pull request merged**